sdks/python/src/opik/evaluation/metrics/llm_judges/g_eval/metric.py

```diff
@@ -0,0 +1,137 @@
+import math
+from typing import Any, Optional, Union
+
+from litellm.types.utils import ModelResponse
+
+from opik.evaluation.metrics import base_metric, score_result
+from opik.evaluation.models import base_model, models_factory
+from opik.logging_messages import GEVAL_SCORE_CALC_FAILED
+from .template import G_EVAL_COT_TEMPLATE, G_EVAL_QUERY_TEMPLATE
+from ... import exceptions
+
+
+class GEval(base_metric.BaseMetric):
+    def __init__(
+        self,
+        task_introduction: str,
+        evaluation_criteria: str,
+        model: Optional[Union[str, base_model.OpikBaseModel]] = None,
+        name: str = "g_eval_metric",
+    ):
+        super().__init__(
+            name=name,
+        )
+        self._init_model(model)
+
+        self.task_introduction = task_introduction
+        self.evaluation_criteria = evaluation_criteria
+
+        prompt = G_EVAL_COT_TEMPLATE.format(
+            task_introduction=self.task_introduction,
+            evaluation_criteria=self.evaluation_criteria,
+        )
+
+        self.llm_chain_of_thought: str = self._model.generate_string(input=prompt)
```

Comment:
If the results are too unstable despite the fact of using log_probs. Maybe it would make sense to generate multiple chains of thought (it can be configurable with some default value) and compute an average score between different scores.